File: 백준/Unrated/1940. 주몽/주몽.py
n = int(input())
m = int(input())
items = list(map(int,input().split()))

items.sort()
left = 0
right = len(items) - 1

answer = 0
while left < right:
    s = items[left] + items[right]
    if s < m:
        left += 1
    elif s > m:
        right -= 1
    else:
        answer += 1
        left += 1
        right -= 1

print(answer)

File: 백준/Silver/1940. 주몽/주몽.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        st = new StringTokenizer(br.readLine());
        int m = Integer.parseInt(st.nextToken());

        int[] nums = new int[n];
        st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }

        Arrays.sort(nums);
        int left = 0;
        int right = n - 1;

        int count = 0;

        while (left < right) {

            if ((nums[left] + nums[right]) < m) {
                left++;
            } else if (nums[left] + nums[right] == m) {
                count++;
                left++;
                right--;
            } else {
                right--;
            }
        }
        System.out.println(count);
    }


}
